### PR TITLE
server side macaroons

### PR DESCRIPTION
**Pull request by zeebo**
What: Adds macaroon support to the server side

Why: So that api keys are now macaroons

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: satellite/console/service.go [hunk 1]
```diff
@@ -651,14 +655,20 @@ func (s *Service) CreateAPIKey(ctx context.Context, projectID uuid.UUID, name st
 		return nil, nil, ErrUnauthorized.Wrap(err)
 	}
 
-	key, err := CreateAPIKey()
+	secret := make([]byte, 32)
```
Comment (rikysya) [inline satellite/console/service.go]: Don't we have a function like `macaroon.NewSecret`? May be we can use it here?

File: satellite/console/service.go [hunk 2]
```diff
@@ -675,14 +676,20 @@ func (s *Service) CreateAPIKey(ctx context.Context, projectID uuid.UUID, name st
 		return nil, nil, ErrUnauthorized.Wrap(err)
 	}
 
-	key, err := CreateAPIKey()
+	secret, err := macaroon.NewSecret()
```
Comment (jtolds) [inline satellite/console/service.go]: oh? a new secret for every key? okay

Comment (zeebo) [reply satellite/console/service.go]: Turns out that's easiest. /shrug

**Comment by zeebo** [comment]
Those are good questions. 

From a UX perspective I could see wanting to show revoked key names in the UI? I dunno. I'm unaware of any guidance on how revocations should work, so I did what seemed obvious to me. Deleting seems more obvious.

It's a small diff to change to either. Happy to defer.

**Comment by jtolds** [comment]
The UX right now is to delete API keys to revoke them, so let's just do the less-LOC approach for now and bring back stuff if we need it

**Comment by thepaul** [comment]
Just to clarify, we will be able to revoke by tail, right? It doesn't seem to make sense by head.

**Comment by zeebo** [comment]
Yeah. The head is just the root-most tail. 💥🤕

**Comment by thepaul** [comment]
Oh. I probably have my terminology wrong. Is tail not the same thing as the HMAC/"signature" part?

**Comment by zeebo** [comment]
Oh yep, you're right. Also, the head is exactly the secret. So yeah unless every macaroon has a distinct secret (as is implemented in this PR), then revoking by the head doesn't make sense.

**Comment by zeebo** [comment]
Nope I'm wrong again. Secret is used twice in the function and I got confused. Head is a random secret unique to each root macaroon, so I think it makes sense to revoke it.

**Comment by zeebo** [comment]
It's equivalent to revoking by the root-most tail, though, so yeah no reason to do anything but tails.

**Comment by jtolds** [comment]
If you want to do the tail thing you can revert this: https://github.com/storj/storj/pull/1862/commits/87f2b67f020d6464a1adae0c1bad984f9f5bbeef

**Comment by zeebo** [comment]
This is ready for a review.

**Pull request merged**